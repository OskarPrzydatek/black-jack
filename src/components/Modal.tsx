import { GameStatusEnum } from '../constants/gameStatus.enum';
import * as S from '../styles/ Modal.style';

import Button from './Button';
import Text from './Text';

interface ModalProps {
  gameStatus: GameStatusEnum;
  gameScore: number;
  dataTestID?: string;
  resultMessageTestID?: string;
  newGameButtonTestID?: string;
  handleNewGame: () => void;
}

export default function Modal({
  gameStatus,
  gameScore,
  dataTestID,
  resultMessageTestID,
  newGameButtonTestID,
  handleNewGame,
}: ModalProps) {
  const gameResultMessage = {
    PersianEye: '!!! PERSIAN EYE !!!',
    Win: `Win with score: ${gameScore}`,
    Pass: `Pass! End with score: ${gameScore}`,
    Lose: 'Game Over',
    Play: '',
  };

  if (gameStatus !== GameStatusEnum.Play)
    return (
      <S.ModalWrapper data-testid={dataTestID}>
        <S.Modal>
          <Text dataTestID={resultMessageTestID}>
            {gameResultMessage[gameStatus]}
          </Text>
          <Button
            dataTestID={newGameButtonTestID}
            label="New Game"
            onClick={handleNewGame}
          />
        </S.Modal>
      </S.ModalWrapper>
    );

  return null;
}
