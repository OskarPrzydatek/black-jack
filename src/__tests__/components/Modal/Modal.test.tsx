import renderer from 'react-test-renderer';
import Modal from '../../../components/Modal';
import { GameStatusEnum } from '../../../constants/gameStatus.enum';

const mockHandleNewGame = jest.fn();

describe('Modal', () => {
  test('component snapshot', () => {
    const tree = renderer
      .create(
        <Modal
          gameStatus={GameStatusEnum.Win}
          gameScore={21}
          handleNewGame={mockHandleNewGame}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
