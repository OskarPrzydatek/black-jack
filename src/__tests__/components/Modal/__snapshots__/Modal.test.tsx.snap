// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapshot 1`] = `
<div>
  <p>
    Win with score: 21
  </p>
  <button
    className="sc-bcXHqe etdubz"
    onClick={[MockFunction]}
  >
    New Game
  </button>
</div>
`;
